app/services/story_service.go

```diff
@@ -69,6 +84,126 @@ func (s *StoryService) CreateStoryForProject(requestData request.CreateStoryRequ
 	return int(createdStory.ID), nil
 }
 
+func (s *StoryService) CreateDesignStoryForProject(file multipart.File, fileName, title string, projectID int) (uint, error) {
+	hashID := s.hashIdGenerator.Generate() + "-" + uuid.New().String()
+	project, err := s.projectService.GetProjectById(uint(projectID))
+	if err != nil {
+		return 0, err
+	}
+	url := "http://localhost:8082/?folder=/workspaces/story" + hashID
+	frontendUrl := "http://localhost:8083/" + project.HashID + "/" + hashID + "/out/"
+	env := config.Get("app.env")
+	if env == "production" {
+		url = "https://" + hashID + ".workspace.superagi.com/?folder=/workspaces/story" + hashID
+		frontendUrl = "https://" + "fe-" + hashID + ".workspace.superagi.com"
+	}
+
+	story := &models.Story{
+		ProjectID:   uint(projectID),
+		Title:       title,
+		Status:      constants.Todo,
+		HashID:      hashID,
+		Url:         url,
+		FrontendURL: frontendUrl,
+	}
+
+	frontendService := "nextjs"
+	//Making Call to Workspace Service to create workspace on project level
+	_, err = s.workspaceServiceClient.CreateFrontendWorkspace(
+		&request.CreateWorkspaceRequest{
+			StoryHashId:      hashID,
+			WorkspaceId:      project.HashID,
+			FrontendTemplate: &frontendService,
+			//FrontendTemplate: &backendService,
+		},
+	)
+
+	if err != nil {
+		fmt.Println("Error creating workspace")
+		return 0, err
+	}
+
+	// create a story
+	createdStory, err := s.storyRepo.CreateStory(story)
+	if err != nil {
+		return 0, err
+	}
+
+	s3Url, err := s.UploadFileBytesToS3(file, fileName, projectID, int(createdStory.ID))
```

Comment:
We need to have localstorage along with s3 upload, refactor out the filesystem part